code/modules/spells/spell_types/knock.dm

```diff
@@ -16,13 +16,16 @@
 	user << sound("sound/magic/Knock.ogg")
 	for(var/turf/T in targets)
 		for(var/obj/machinery/door/door in T.contents)
-			spawn(1)
-				if(istype(door,/obj/machinery/door/airlock))
-					door:locked = 0
-				door.open()
+			addtimer(src, "open_door", 1, FALSE, door)
 		for(var/obj/structure/closet/C in T.contents)
-			spawn(1)
-				C.locked = 0
-				C.open()
+			addtimer(src, "open_closet", 1, FALSE, C)

```

Comment:
Spawns like these should be replaced with set waitfor = 0 procs, as the spawn is about avoiding the sleep in open() sleep.